docs/object_code.md

```diff
@@ -0,0 +1,272 @@
+# Exo Object Code Syntax
```

Comment:
As we discussed, should bring up array vs window parameter syntax (`f32[...]` vs `[f32][...]`).